added cell delete button on profile page

836hardik-agrawal opened the pull request:
**Changes Made**
`Backend (backend/api)`
1.`models/cell.py` – Added support for deleting a cell from the database.
2.`resources/cell.py` – Implemented a delete function to handle requests from the frontend.
`Frontend (frontend/src)`
1.`pages/profile/component/cellList.jx` – Integrated DeleteCellModal for user interaction.
2.`pages/profile/component/DeleteCellModal.jsx` – Created a modal to accept cellId as a parameter for deletion.
3.`services` – Implemented a function to call the backend API for deletion.
**Testing & Validation**
Verified delete functionality with valid inputs.
Confirmed that UI updates reflect deletions immediately.

836hardik-agrawal commented:
Fixes [#394 ](https://github.com/jlab-sensing/ENTS-backend/pull/394/commits)

jmadden173 reviewed (commented):
Hi @836hardik-agrawal, I triggered the actions. The linting errors will need to be fixed before it can be merged?

Can you also add a screenshot to the PR showing what the delete button looks like?

I saw that you did some tests on the endpoint. Have you checked that when a cell is deleted, the associated rows in `data` and `sensor` table also get removed?

Also deleting cells is something that should happen infrequently, and I think that it would be worth it to have a "Are you sure?" popup before actually deleting the cells to prevent accidental deletions. Is this something that would be difficult to implement?

836hardik-agrawal commented:
Hi @jmadden173 
Thanks for your detailed feedback.
I have presently resolved the lint issues,with the files which I have changed .
Also shortly I will make a PR doing all the changes suggested along with video demonstration .
Does I need to also add test coverage ?

jmadden173 commented:
I don't see the changes you made on this PR. You shouldn't need to open a new one. And yes you will likely have to add tests for the delete button to get the codecov checks passing. It may be annoying but helps when multiple people are contributing to ensure that changes someone else makes doesn't break implementations in the code you add.

836hardik-agrawal commented:
![img1](https://github.com/user-attachments/assets/0889d793-94a3-41b4-b88e-4111c000cdb8)
![img2](https://github.com/user-attachments/assets/cbda19b6-9c0a-4d79-b3cd-15e47056f944)
![img3](https://github.com/user-attachments/assets/ff2a4fda-7b5d-4585-86d5-0a6f949f1e50)
![img4](https://github.com/user-attachments/assets/895c63ac-c523-4560-a9c4-b409ae241a19)

I have formatted the code and solved lint issue on code I have added.
![check_file](https://github.com/user-attachments/assets/71bb39f0-9c5e-4a92-9502-cd568c152fdf)
I also made sure entries get deleted in data and sensor table.
Also I added the "Are you sure" before deleting.
Also I add test case for delete button.

Please review it when free
@jmadden173

jmadden173 commented:
Triggered the lint/test action. I believe you forgot to run the linter on the `frontend/` folder. I'm going to pass it off to @aleclevy to take a look at the UI elements.

aleclevy commented:
Can you fix the linting error on the frontend?

836hardik-agrawal commented:
Sorry for late response . I will fix it by today .

836hardik-agrawal commented:
![image](https://github.com/user-attachments/assets/f93941c8-765a-45d6-b205-02a4d8898f05)
Sorry but when I fix the linting issue . I get this error . I am not getting how to correct it . 
@aleclevy

jmadden173 commented:
Bump on this @aleclevy
